cmd/metacache-server-pool.go

```diff
@@ -449,5 +440,10 @@ func (z *erasureServerPools) listAndSave(ctx context.Context, o *listPathOptions
 		xioutil.SafeClose(saveCh)
 	}()
 
-	return filteredResults()
+	entries, err = filteredResults()
+	if err == nil && err != io.EOF {
```

Comment:
isn't this OR rather than AND ?

Reply:
Yeah, it is a bit superfluous. Probably it should be `|| err == io.EOF`.

Reply:
Let's check it.. Adds some extra safety.